src/processor/StructureDefinitionProcessor.ts

```diff
@@ -42,8 +46,9 @@ export class StructureDefinitionProcessor {
         elements,
         existingInvariants
       );
+      const mappings = StructureDefinitionProcessor.extractMappings(elements, input, fhir);
       StructureDefinitionProcessor.extractRules(input, elements, sd, fhir);
-      return [sd, ...invariants];
+      return [sd, ...[...invariants, ...mappings]];
```

Comment:
```suggestion
      return [sd, ...invariants, ...mappings];
```
I think that should work, shouldn't it?

Reply:
I've been struggling over this line. If I leave the type that we return for this function as it is, I don't think your suggested change works, although I really don't understand why. I believe that the type we are declaring is designed so that we know the first element should be either an `ExportableProfile` or `ExportableExtension`. If that is the case and we do want to ensure that the first element is a profile or extension, I'm not sure I know what the right way to have the rest of the elements be either `ExportableInvariants` or `ExportableMappings`. If we aren't intending to have the type of the first element be specifically a Profile or Extension, then I think we should update the return type of the function, which would let me use the simpler suggestion you have here.

But again, I'm getting fairly confused by TypeScript here, so I'm definitely willing to believe I'm missing something.

Reply:
OK.  If my suggestion doesn't work as-is then I am also confused by typescript, and in that case I don't really have any suggestions.  If what you did works, then it is fine.  I just thought it was weird -- but I guess it is _necessarily_ weird.

Reply:
a4901fe adds a TODO to use this much clearer version of this line when we use TypeScript 4.0